src/toolchain.rs

```diff
@@ -304,7 +308,7 @@ impl Toolchain {
         let fuelup_bin = fuelup_bin();
         if !fuelup_bin.is_file() {
             info!("fuelup not found - attempting to self update");
```

Comment:
This doesn't seem like it should be possible 🤣 If someone wanted to install it only from cargo, this would not update the cargo version, and install fuelup in the home directory, which is probably not what they want. 

In that case I think we should give an error message like `fuelup cannot update itself because it was installed by a package manager. For full functionality, visit https://docs.fuel.network/docs/fuelup/installation and re-install.`